### ISSUE TITLE
Bug in neural network training with gradients

### ISSUE THREAD

**Issue by frostedoyster** [lab-cosmo/sphericart#59]
The derivatives of the gradient errors with respect to the weights are computed, but it seems that one or more terms are missing or incorrect. A temporary fix is to use `backward_second_derivatives=True` at class initialization.
